File: aat-android/src/main/java/ch/bailu/aat/map/layer/grid/Crosshair.java
package ch.bailu.aat.map.layer.grid;

import org.mapsforge.core.model.Point;

import ch.bailu.aat_lib.map.MapContext;
import ch.bailu.aat_lib.map.layer.MapLayerInterface;
import ch.bailu.aat_lib.preferences.StorageInterface;

public final class Crosshair implements MapLayerInterface {
    @Override
    public void onLayout(boolean changed, int l, int t, int r, int b) {

    }

    @Override
    public void drawInside(MapContext mcontext) {

    }

    @Override
    public void drawForeground(MapContext mcontext) {
        drawGrid(mcontext);



    }


    private void drawGrid(MapContext mcontext) {
        final ch.bailu.aat_lib.map.Point pixel = mcontext.getMetrics().getCenterPixel();

        mcontext.draw().vLine(pixel.x);
        mcontext.draw().hLine(pixel.y);
        mcontext.draw().point(pixel);
    }



    @Override
    public boolean onTap(Point tapPos) {
        return false;
    }

    @Override
    public void onPreferencesChanged(StorageInterface s, String key) {}


    @Override
    public void onAttached() {

    }

    @Override
    public void onDetached() {

    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/grid/Crosshair.kt
package ch.bailu.aat_lib.map.layer.grid

import ch.bailu.aat_lib.map.MapContext
import ch.bailu.aat_lib.map.layer.MapLayerInterface
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.util.Point

class Crosshair : MapLayerInterface {
    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {}
    override fun drawInside(mcontext: MapContext) {}
    override fun drawForeground(mcontext: MapContext) {
        drawGrid(mcontext)
    }

    private fun drawGrid(mcontext: MapContext) {
        val pixel = mcontext.getMetrics().getCenterPixel()
        mcontext.draw().vLine(pixel.x)
        mcontext.draw().hLine(pixel.y)
        mcontext.draw().point(pixel)
    }

    override fun onTap(tapPos: Point): Boolean {
        return false
    }

    override fun onPreferencesChanged(storage: StorageInterface, key: String) {}
    override fun onAttached() {}
    override fun onDetached() {}
}
